Boost: Migrate svelte components to react

dilirity opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Migrate several components to React and added classname pnpm dependency.

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* ActivateLicense
* ImageCdnRecommendation
* Image Size Analysis - Hero
* RecommendationContext (tooltip)
* ScoreContext (tooltip)
* Image Size Analysis - OtherGroupContext (tooltip)
* Support

Tooltips throughout the UI now use the IconTooltip RNA component.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Check if the UI components work as expected.

dilirity committed:
Migrate Support component

dilirity committed:
Migrate ActivateLicense component

dilirity committed:
Merge branch 'trunk' into boost-react/migrate-components

dilirity committed:
Update ActivateLicense component structure

dilirity committed:
Add section component

dilirity committed:
Update Support component structure

dilirity committed:
Remove section component

dilirity committed:
Remove (svelte) ActivateLicense component

dilirity committed:
Migrate (react) ImageCDNRecommendation component

dilirity committed:
Migrate (react) RecommendationContext component

dilirity committed:
Migrate (react) RecommendationContext component

dilirity committed:
add changelog

dilirity committed:
Migrate (to react) ISA Hero component

dilirity committed:
Simplify render check

dilirity committed:
Merge branch 'trunk' into boost-react/migrate-components

dilirity committed:
Migrate ScoreContext component

dilirity committed:
Update recommendation context component to use IconTooltip

dilirity committed:
Refactor OtherGroupContext component

dilirity committed:
Remove icon tooltip styles

dilirity committed:
Fix hard-coded color

dilirity committed:
Fix score context tooltip alignment

dilirity committed:
Remove unnecessary text wrapper

dilirity committed:
Update recommendation context tooltip alignment

dilirity committed:
Add fade in transition for ISA hero component

dilirity committed:
Update plugin version (make changelogger happy)

dilirity commented:
> Speed Score feature › Should be able to hover info icon next to overall score and see the detailed overall score description popin 

Tests are broken. Info tooltip no longer works on hover - just on mouse click.

haqadn commented on projects/plugins/boost/app/assets/src/js/react-components/image-size-analysis/hero/index.tsx:
```diff
@@ -0,0 +1,98 @@
+	return (
+		<div
+			className={ classNames( styles.hero, {
+				[ styles[ 'fade-in' ] ]: fadeProp,
```
I think the fade-in effect can be achieved purely with CSS using animation. So, I don't feel very comfortable involving the state in this. Can you try replacing it with [CSS animation](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_animations/Using_CSS_animations)?

haqadn replied:
Here is an implementation just in case: https://github.com/animate-css/animate.css/blob/main/source/fading_entrances/fadeIn.css

dilirity replied:
Oh wow! I tried replicating the svelte way via JS and didn't think to use CSS. Thanks!

haqadn commented on projects/plugins/boost/app/assets/src/js/react-components/recommendation-context/index.tsx:
```diff
@@ -0,0 +1,18 @@
+import { ImageCdnRecommendation } from '../image-cdn-recommendation';
+import styles from './styles.module.scss';
+
+export const RecommendationContext = () => {
```
Considering it is a simple usage of the IconTolltip, I think we can skip putting it in a separate component and place the IconTolltip directly in `Hero`.

dilirity replied:
This makes sense - removed!

haqadn commented on projects/plugins/boost/app/assets/src/js/react-components/image-size-analysis/other-group-context/index.tsx:
```diff
@@ -0,0 +1,29 @@
+import { __ } from '@wordpress/i18n';
+import styles from './styles.module.scss';
+
+const OtherGroupContext = () => {
```
I think we do not need this component either, just like `RecommendationContext`. But, I think you are using it as a stop-gap until `MultiProgress` is refactored into a react component, in which case, it is ok.

dilirity replied:
This will be moved when refactoring MultiProgress, as it holds styles. I've added a todo in MultiProgress about this.

haqadn commented on projects/plugins/boost/app/assets/src/js/react-components/score-context/index.tsx:
```diff
@@ -0,0 +1,57 @@
+import { __ } from '@wordpress/i18n';
+import styles from './styles.module.scss';
+
+const ScoreContext = () => {
```
Same as `RecommendationContext` and `OtherGroupContext`.

dilirity replied:
This will be removed when refactoring MultiProgress, as it holds styles. I've added a todo in MultiProgress about this.

dilirity replied:
I've also included a todo for this one as well.

haqadn reviewed (changes requested):
This covers a lot and it looks like it's working well. My recommendation is to organize the files in the proposed structure in the p2. And, I think non-reusable components like Tooltips can live in their parent component folder.

I have noticed a small issue in styles, the icon placement is wrong:

This PR:
![image](https://github.com/Automattic/jetpack/assets/3737780/17b7090e-4393-45ec-a516-f669d15cfbc6)

Trunk:
![image](https://github.com/Automattic/jetpack/assets/3737780/966b04ff-03f3-4559-8e3f-0a47f4ba5468)

haqadn commented:
I think this also breaks an e2e test since `IconTooltip` from RNA needs a click instead of a hover.

dilirity committed:
Update component structure and usage

dilirity committed:
Move recommendation context logic into parent component

dilirity committed:
Add todo for other-group-context

dilirity committed:
Add todo for score-context

dilirity committed:
Update hero fade in to use CSS animation

dilirity committed:
Fix score tooltip icon alignment

dilirity committed:
Attempt to fix e2e tests

haqadn reviewed (changes requested):
A couple of suggestions related to organization:

1. Instead of  `react-components/image-size-analysis/hero` we can have `react-components/sections/isa-hero`, which conforms to the flat structure.
2. `react-components/components/score-context` seems only useful for the future "score" section. In which case, it should live under something like `react-components/sections/score/context-tooltip.tsx`.

 
My thoughts regarding sections and components are: 
sections = Single use parts of UI
components = Generic UI elements which can be used by multiple sections
pages = What we mount as a route in navigation

What do you think?

dilirity committed:
Merge branch 'trunk' into boost-react/migrate-components

dilirity committed:
Update react ISA related component structure

dilirity committed:
Update score context location

dilirity commented:
@haqadn I've addressed the suggestions - thank you for those. I'll leave my comments about the structure in the P2, so we have the conversation in one place.

dilirity committed:
Merge branch 'trunk' into boost-react/migrate-components

dilirity committed:
Merge branch 'trunk' into boost-react/migrate-components

haqadn reviewed (approved):
:shipit:

dilirity committed:
Merge branch 'trunk' into boost-react/migrate-components
